File: TopCoder/src/main/java/com/leetcode/String/CountAndSay.java
package com.leetcode.String;

public class CountAndSay {
    public String countAndSay(int n) {
        String s1 = "1";
        for (int i = 2; i <=n ; i++) {
            char c1 = s1.charAt(0);
            int count = 1;
            StringBuffer stringBuffer = new StringBuffer();
            for ( int j = 1 ; j < s1.length() ; j++) {
                if (s1.charAt(j) == c1) {
                    count++;
                } else {
                    stringBuffer.append(count+(c1+""));
                    c1 = s1.charAt(j);
                    count = 1;
                }
            }
            stringBuffer.append(count+(c1+""));
            s1 = stringBuffer.toString();
        }
        return s1;
    }

    public static void main(String[] args) {
        int n = 5;
        String res = new CountAndSay().countAndSay(n);
        System.out.println(res);
    }
}

File: TopCoder/python/misc/CountAndSay.py
class Solution:
    def countAndSay(self, n: int) -> str:
        s1 = "1"
        while n > 1:
            count = 1
            string_builder = ""

            string_length = 1

            while string_length <= len(s1):
                if string_length == len(s1) or s1[string_length] != s1[string_length-1]:
                    string_builder = string_builder + str(count) + s1[string_length-1]
                    count = 1
                else:
                    count = count + 1
                string_length = string_length + 1
            s1 = string_builder
            n = n - 1
        return s1

if __name__ == "__main__":
    sol = Solution()
    res = sol.countAndSay(3)
    print(res)